### ISSUE TITLE
documentation for migrate:import --progress option is unclear

### ISSUE THREAD

**Issue by joachim-n** [drush-ops/drush#5464]
**Describe the bug**

`drush help mim` says: 

```
 --progress[=PROGRESS]  Show progress bar [default: true]
```

Reading this, I assumed I'd say ` --progress=false` to turn it off, since it gives 'true' as the default. However that didn't work. What did work as ` --progress=0`

Also, the examples have:

```
--no-progress
```

which isn't given in the list of options.

The code only has this:

```
     * @option progress Show progress bar
```

So this looks to me like a bug in the way command help documentation is derived from the code.

**To Reproduce**
What did you do?

**Expected behavior**
What did you expect would happen?

**Actual behavior**
What happened instead?

**Workaround**
Is there another way to do the desired action?

### System Configuration
| Q               | A
| --------------- | ---
| Drush version?  | 11.x/10.x/8.x (please be specific, and try latest release)
| Drupal version? | 10.x/9.x/8.x/7.x
| PHP version     | 8.x/7.x
| OS?             | Mac/Linux/Windows

**Additional information**

**Comment by claudiu-cristea** [comment]
@joachim-n , this is how it suppose to work. Please see https://github.com/consolidation/annotated-command#option-default-values, the _The special value true produces the following behavior:_ section

**Comment by claudiu-cristea** [comment]
Particularly...

> If the user adds `--no-foo` on the commandline, then the value of `foo` will be `false`.

As `--no-progress` is documented in the usage examples, it should be enough. Or should we extend the docs of `@option progress`?

**Comment by joachim-n** [comment]
> Or should we extend the docs of @option progress?

Extend, and fix, I'd say.

```
--progress[=PROGRESS]
```
The [=PROGRESS] makes no sense here, since you give this option without a value.

And it should say something like 'enabled by default. Negate with --no-progress'.

**Comment by claudiu-cristea** [comment]
> The [=PROGRESS] makes no sense here, since you give this option without a value

That is added automatically. How to get rid of?

**Comment by joachim-n** [comment]
That's what I mean -- the code that adds this automatically has a bug.

**Comment by claudiu-cristea** [comment]
But is that code Drush? I think it's https://github.com/consolidation/annotated-command

**Comment by weitzman** [comment]
I looked into this a bit. The default value of `true` does accept a string, so the [=PROGRESS] is rightly shown. If we dont want a value, I'm not sure that annotated-command provides a good option for us. In 12.x branch, we could potentially expand the Option Attribute to indicate \Symfony\Component\Console\Input\InputOption::VALUE_NONE.

**Comment by greg-1-anderson** [comment]
We might also be able to improve behavior a bit in Drush 12 if we use InputOption::VALUE_NEGATABLE. I haven't looked into supporting this in Annotated Commands much yet.

**Comment by weitzman** [comment]
Related https://github.com/drush-ops/drush/issues/4825#issuecomment-918695596

**Comment by joachim-n** [comment]
> The default value of true does accept a string, so the [=PROGRESS] is rightly shown

Indeed, but it's confusing that specifying '=false' counts as truthy!
